Automattic/jetpack
# Packages: Add a basic Jetpack Logo package

This PR is an experiment to create and use a Jetpack Logo as a composer package. Similar to #12363, but I wanted to start it from scratch to go through the process, plus I introduced several changes, some of which are opinionated.

This is an initial attempt to address 3879-gh-jpop-issues.

I'm planning to use this PR for a base of several other experiments too. Feedback is appreciated.

**Note on design review**: The only design feedback I'm looking here is the alt attribute change we're doing here, in favor of better accessibility. See [convo](https://github.com/Automattic/jetpack/pull/12379#discussion_r286237047) for details.

#### Changes proposed in this Pull Request:

* Same as #12363, but has several more changes:
* **Design changes:** Changed the original `alt` attribute to just `Jetpack.` - see [convo](https://github.com/Automattic/jetpack/pull/12379#discussion_r286237047)
* Source directory renamed to `/src` instead of `/lib`, because it's more common for PHP libraries.
* Package dir renamed to `/logo` instead of `/jetpack-logo`, because `jetpack` is assumed and therefore arbitrary.
* Removed version from the namespace, because I think it can cause more problems than it would actually solve (also see this discussion: https://github.com/Automattic/jetpack/pull/12213#discussion_r281981451).
* Rename `utils` in the namespace to `assets`, because it's more specific.
* Rename class to just `Logo` - Jetpack already exists everywhere, including in the namespace.
* Fixed the license - now it's the same as Jetpack, the plugin.
* Refactored a bit to make easier to test and more flexible.
* Move the actual logo file to the package.
* Allow for custom logo URL in the class interface.
* Add a README with example usage.
* Ignore composer.lock in packages
* Add packages directory to the phpcs whitelist for linting.
* Add some temporary phpcs ignores - in the long term, we need to make PSR-4 and WP coding standards compatible.
* Added basic tests for the new package.
* Add `composer install` to the Travis CI setup.
* **Bump the minimum PHP version to 5.3.2** because of Composer's minimum requirements.
* Add a `php-build` script that basically runs `composer install` and add it to our build scripts.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
See p1HpG7-6V2-p2 and p1HpG7-6Ym-p2 for details.

#### Testing instructions:
* Make sure you're not connected to WP.com, but Jetpack is activated.
* Go to /wp-admin/plugins.php
* Make sure the logo is displayed properly.
* Run the tests and make sure they pass: `yarn docker:phpunit --testsuite packages`

#### Proposed changelog entry for your changes:
* Introduced a new Jetpack Logo package.

#### Notes
* Architecture of how tests are structured is likely going to change soon. While they're still in the global directory, we're going to move them out to the packages and make them run both independently and together as a whole. This should be good for now though.
* We'll have to build a fresh `composer.lock` after releasing the logo as a package.
